spec/features/schools/register_mentor/edge_cases/already_active_at_school_after_trn_spec.rb

```diff
@@ -33,7 +33,8 @@ def given_there_is_a_school_in_the_service
   end
 
   def and_there_is_an_ect_with_no_mentor_registered_at_the_school
-    @ect = FactoryBot.create(:ect_at_school_period, :active, school: @school)
+    lead_provider = FactoryBot.create(:lead_provider, name: "Xavier's School for Gifted Youngsters")
+    @ect = FactoryBot.create(:ect_at_school_period, :active, lead_provider:, school: @school)
```

Comment:
Should we always associate a `lead_provider` in the `:ect_at_school_period` factory instead?
Or at least a `trait :with_lead_provider`? 
Otherwise, we are repeating this code many times.

Reply:
I have done that already in another branch, here it is still appropriate, for the name to be checked